public/app/features/explore/Logs/LogsContainer.tsx

```diff
@@ -208,18 +209,33 @@ class LogsContainer extends PureComponent<LogsContainerProps, LogsContainerState
   };
 
   showContextToggle = (row?: LogRowModel): boolean => {
-    if (!row?.dataFrame.refId || !this.state.logContextSupport[row.dataFrame.refId]) {
+    if (!row?.dataFrame.refId || !this.state.dsInstances[row.dataFrame.refId]) {
       return false;
     }
-
-    return true;
+    return hasLogsContextSupport(this.state.dsInstances[row.dataFrame.refId]);
   };
 
   getFieldLinks = (field: Field, rowIndex: number, dataFrame: DataFrame) => {
     const { splitOpenFn, range } = this.props;
     return getFieldLinksForExplore({ field, rowIndex, splitOpenFn, range, dataFrame });
   };
 
+  logDetailsFilterAvailable = () => {
+    return Object.values(this.state.dsInstances).some((ds) => ds?.modifyQuery || hasToggleableQueryFiltersSupport(ds));
```

Comment:
Great question. Yes and no. Checking for `ds?.modifyQuery` is the backwards compatible way that covers both: the original implementation using `modifyQuery` and those who implement the new interface.

https://github.com/grafana/grafana/blob/main/public/app/features/explore/Logs/LogsContainer.tsx#L93

It could be

```suggestion
    return Object.values(this.state.dsInstances).some((ds) => ds?.modifyQuery || hasQueryModificationSupport(ds) || hasToggleableQueryFiltersSupport(ds));
```

But seems redundant. When we fully deprecate `modifyQuery` from `DataSourceAPI` we can update this.

Reply:
I think I would still go the more redundant way as it's easier to followup in the future. WDYT?

Reply:
It certainly doesn't hurt.